realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -437,7 +442,16 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
                 // refresh the Realm to the version provided by the worker thread
                 // (advanceRead to the latest version may cause a version mismatch error) preventing us
                 // from importing correctly the handover table view
-                realm.sharedGroupManager.advanceRead(result.versionID);
+                try {
+                    realm.sharedGroupManager.advanceRead(result.versionID);
+                } catch (BadVersionException e) {
```

Comment:
Yes, it should be safe, but we should actually abort here. I forgot to `return` if this is thrown. If this happens, a REALM_CHANGED event should be in the Looper queue anyway which will schedule a retry. Until that happens it will just continue using the old version (which should be fine).